python/pyspark/sql/tests/connect/test_connect_function.py

```diff
@@ -645,6 +645,153 @@ def test_string_functions(self):
             sdf.select(SF.encode("c", "UTF-8")).toPandas(),
         )
 
+    # TODO(SPARK-41283): To compare toPandas for test cases with dtypes marked
+    def test_date_ts_functions(self):
+        from pyspark.sql import functions as SF
+        from pyspark.sql.connect import functions as CF
+
+        query = """
+            SELECT * FROM VALUES
+            ('1997/02/28 10:30:00', '2023/03/01 06:00:00', 'JST', 1428476400, 2020, 12, 6),
+            ('2000/01/01 04:30:05', '2020/05/01 12:15:00', 'PST', 1403892395, 2022, 12, 6)
+            AS tab(ts1, ts2, tz, seconds, Y, M, D)
+            """
+        # +-------------------+-------------------+---+----------+----+---+---+
+        # |                ts1|                ts2| tz|   seconds|   Y|  M|  D|
+        # +-------------------+-------------------+---+----------+----+---+---+
+        # |1997/02/28 10:30:00|2023/03/01 06:00:00|JST|1428476400|2020| 12|  6|
+        # |2000/01/01 04:30:05|2020/05/01 12:15:00|PST|1403892395|2022| 12|  6|
+        # +-------------------+-------------------+---+----------+----+---+---+
+
+        cdf = self.connect.sql(query)
+        sdf = self.spark.sql(query)
+
+        # With no parameters
+        for cfunc, sfunc in [
+            (CF.current_date, SF.current_date),
+        ]:
+            self.assert_eq(
+                cdf.select(cfunc()).toPandas(),
+                sdf.select(sfunc()).toPandas(),
+            )
+
+        # current_timestamp
+        # [left]:  datetime64[ns, America/Los_Angeles]
+        # [right]: datetime64[ns]
+        plan = cdf.select(CF.current_timestamp())._plan.to_proto(self.connect)
+        self.assertEqual(
+            plan.root.project.expressions.unresolved_function.function_name, "current_timestamp"
+        )
+
+        # localtimestamp
+        plan = cdf.select(CF.localtimestamp())._plan.to_proto(self.connect)
+        self.assertEqual(
+            plan.root.project.expressions.unresolved_function.function_name, "localtimestamp"
```

Comment:
Cannot compare with PySpark since we cannot call them at the exact same timestamp.

Reply:
see above